substrate/frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -1862,6 +1865,99 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn mine_solution_always_respects_max_backers_per_winner() {
```

Comment:
great test!

Reply:
I do strongly think though if you move the trimming to this file, and do it post-phragmen, it will be less error-prone: 

Let phragmen, which is a very old legacy code do its thing. Once the result is back, if any winner has more backers than we wanted, we will sort their backers and remove the least staked ones. 

We already do some trimming in this file based on weight/length. This will be another axis. 

I do think that this approach is cleaner (all trimming happens in one stage) and less error prone (no legacy code is changed).

That being said, I have not seen the details of https://github.com/paritytech/polkadot-sdk/pull/6482 yet, maybe it is a trivial change.

Reply:
@kianenigma I reverted the changes introduced before in Phragmen and implemented the max backers per winner trimming at the miner level https://github.com/paritytech/polkadot-sdk/pull/6771.